### PR TITLE
Add error to spans

### LINKED ISSUE #86: Add the error attribute as a first-class option for Spans
If you set an the attribute `error` to `true` in a span, then the DT UI knows it was an span with an error in it. we should provide that as a first-class option in our Span builder.

### PR DESCRIPTION
**Pull request by breedx-nr**
This resolves #86 .

### CODE REVIEW

File: telemetry_core/src/main/java/com/newrelic/telemetry/spans/Span.java [hunk 1]
```diff
@@ -146,9 +150,15 @@ public SpanBuilder parentId(String parentId) {
       return this;
     }
 
+    public SpanBuilder isError() {
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/spans/Span.java]: javadoc, por favor!

File: telemetry_core/src/main/java/com/newrelic/telemetry/spans/Span.java [hunk 2]
```diff
@@ -146,9 +150,14 @@ public SpanBuilder parentId(String parentId) {
       return this;
     }
 
+    public SpanBuilder isError(){
```
[the remark this answers is not in the document]
Comment (breedx-nr) [reply telemetry_core/src/main/java/com/newrelic/telemetry/spans/Span.java]: I like it, I think the name isn't great as it is now.  Will change!

File: telemetry_core/src/main/java/com/newrelic/telemetry/spans/Span.java (hunk 1 above)
Comment (breedx-nr) [reply telemetry_core/src/main/java/com/newrelic/telemetry/spans/Span.java]: Yes.

**Comment by breedx-nr** [comment]
Ready for another  👁 👁 ···

**Pull request merged**